net/fetcher.go

```diff
@@ -15,7 +15,7 @@ import (
 
 // Fetcher defines an interface that may be used to fetch data from the network.
 type Fetcher interface {
-	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, recur int) ([]types.TipSet, error)
+	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, done func(ts types.TipSetKey) bool) ([]types.TipSet, error)
```

Comment:
Operating over a TipSet**Key** here may prove problematic as it doesn't allow for attributes like `Height` to be inspected. We may want to change this to a full TipSet later on but currently there is no reason to.

Another option might be to provide another function as a parameter that is called after the tipset at tsKey has been fetcher.

Reply:
As noted above, I think *only* the latter is needed.